### PR TITLE
modified homepage

### PR DESCRIPTION
**Pull request by amanpatel123**
Related Merge: #156
Closes: #153 

### What are you trying to accomplish?
Changing the homepage into one complete Markdown into separate markdowns of events

### How are you accomplishing it?
- We already had MarkdowmPages Table so thought I can just reuse it
- Kaminari for pagination

### Is their anything reviewers should know?
So currently in production, the is only one record in the MarkdownPage table which is the long Markdown page with all the news since day 1. You must have seen that after merging this we will need to change the prod data a little bit. 
Question is:
- Should the data be populated before we merge this (Which will require adding of data_migration gem) or
- Or after its merged, we can separate the news and start adding it from lowest first sequence (Since the news are being sorted based on `created_at` pattern ) 

Depending on what @EricPickup suggest I am fine with anyone 

- [x] Is it safe to rollback this change if anything goes wrong?

### CODE REVIEW

**Comment by amanpatel123** [comment]
@EricPickup how does this look now?

**Comment by amanpatel123** [comment]
Ouu I did not see that they are being used somewhere. Sorry about that

**Comment by amanpatel123** [comment]
After looking at the code, I was wondering do we really need pageController and views? Pages are indirectly rendering Markdown pages. Meaning having those views directly as markdown pages would suffice. If we think about it, `Home`, `discord`, `guide`, `about` are all markdown pages and could be just handled by Markdown_controller. Would love to hear your thoughts about it @EricPickup

**Comment by EricPickup** [comment]
Makes sense to me @amanpatel123, could be something to do in another PR

**Comment by amanpatel123** [comment]
@EricPickup would you like to have a look once again? As we discussed I just migrated all the markdown_pages under markdown_page resources and removed pages controller and views.

**Comment by amanpatel123** [comment]
So first things first, 

>  For example, on any other page besides the Home page I have the option to create a new "page" but realistically nothing happens when we do that because these regular markdown pages are meant to be one body of text and only loads one MarkdownPage.

About this comment, The reason nothing happens when you add a new markdown page is not because of the way markdown is rendered. The reason is for all other pages apart from Home page I use `find_by` to search for the `Markdown` for that specific page. For the home page, I use `where`. Take a look 

```ruby
  def index
    @page_title = "Home"
    @pages = MarkdownPage.where(title: @page_title).order(created_at: :desc).page(params[:page])
  end

  def about
    @page_title = "About"
    @page = MarkdownPage.find_by(title: @page_title)
  end
```
This was done on purpose since this issue only wants us to be able to add and render more than one MarkdownPages to the Home page and none other pages.

Let's go to your second example now: 

> Another example is that with a Post, we'd like to have titles built into it. Right now it's set up so that we're creating multiple MarkdownPages with the same title "Home" and it's making it weird and confusing. If I change that title from "Home" to anything else that post is created but not shown anywhere.

Yes absolutely correct that it's super confusing to rely on title. I have a fix for that in mind and I was supposed to fix that in next MR. This was how it was set earlier and is not something that this issue was concerned about.  I only wanted to focus on the issue and none other fixes. I hope that makes sense? 

They honestly don't see them as 2 different things and in fact, this was an attempt to make it less confusing. It's not that I did not think about your idea of creating a Post model. I just felt that we can still keep all our pages a collection of `Markdowns` and still be able to attain the same functionality. 

I like the idea of how each page is simply rendering a markdown page: 
```ruby  
<%= render 'markdown_pages/markdown_page', page: @page, page_title: @page_title %>
```

Post model with `title`, `content`, and `created_at` is nothing but an another form of saying a `MarkdownPage` model with `title`, `text` and `created_at`. Correct me if I am wrong but when you say my `index` page is a collection `Post`  is nothing but simply a collection of `Markdowns`. I definitely feel that this can avoid having the new button on other pages apart from `Home` but that was the reason for using `find_by`. We can hide the new button from other pages if that's what's confusing.


I think these are the two approaches of solving and just depend on which one is the more preferred way of doing it.  If it still looks confusing then I will be happy to make Post model.

**Comment by EricPickup** [comment]
@amanpatel123 I can see your argument. I guess it comes down to personal preference. You're the  dev of the website now so feel free to do it whichever way you prefer!

**Comment by amanpatel123** [comment]
@EricPickup I think I would love to know if I am incorrect here. All point of asking you to review my PR is to improve my self. If possible I would always love to have your comments on my PR's

**Comment by EricPickup** [comment]
@amanpatel123 neither way is incorrect imo. Once you're completely done with your plans for this then it will be fine. It just seems incomplete right now since there's a couple outstanding things that could be done to improve it which is why I had concerns. It's better now that I know that you know of these issues and plan on fixing it. It just seems like you want to push your changes iteratively, i.e. ship this out and then patch up the last couple things in separate PRs. I personally prefer to ship it entirely finished.

> This was done on purpose since this issue only wants us to be able to add and render more than one MarkdownPages to the Home page and none other pages.

I understand that, but if you're going this route then you should get rid of the option to add a new page on all pages besides Home if one already exists (since it effectively doesn't and shouldn't work).

> Yes absolutely correct that it's super confusing to rely on title. I have a fix for that in mind and I was supposed to fix that in next MR. This was how it was set earlier and is not something that this issue was concerned about. I only wanted to focus on the issue and none other fixes. I hope that makes sense?

For future reference you should mention this in your PR description or somewhere so reviewers will know that you have future plans to improve/fix it

File: app/views/markdown_pages/discord.html.erb [hunk 1]
```diff
@@ -16,7 +16,15 @@
 <center><%= image_tag 'https://discordapp.com/api/guilds/453373412707008522/widget.png' %></center>
 <br>
 
+<% if @page.present? %>
   <%= render 'markdown_pages/markdown_page', page: @page, page_title: @page_title %>
+<% else %>
+  <p>There is no content to show currently</p>
```
Comment (Prakort) [inline app/views/markdown_pages/discord.html.erb:22]: @amanpatel123 We don't need this for this view. 
Screen Shot 2020-05-16 at 11 12 48 AM

Comment (amanpatel123) [reply app/views/markdown_pages/discord.html.erb:22]: okay! will remove it

**Pull request merged**